### ISSUE TITLE
Call to a member function has() on null

### ISSUE THREAD

**Issue by shubaivan** [schmittjoh/JMSDiExtraBundle#307]
U use Symfony 3.3.10 and php 7.1. I install `"jms/di-extra-bundle": "1.7.0"` and after that have error
```
Call to a member function has() on null
vendor/symfony/symfony/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerTrait.php
line 356
```
my action 

```
    public function getAdminQuestionAction(ParamFetcher $paramFetcher)
    {
        try {
            $em = $this->getDoctrine();
```

after when I called `getDoctrine` I have exception 
and code where throw exception, line 356

```
    /**
     * Shortcut to return the Doctrine Registry service.
     *
     * @return Registry
     *
     * @throws \LogicException If DoctrineBundle is not available
     */
    protected function getDoctrine()
    {
        if (!$this->container->has('doctrine')) {
            throw new \LogicException('The DoctrineBundle is not registered in your application.');
        }

        return $this->container->get('doctrine');
    }
```

**Comment by zhil** [comment]
related symfony2 issue https://github.com/symfony/symfony/issues/23200

**Comment by zhil** [comment]
this issue looks like duplicate of https://github.com/schmittjoh/JMSDiExtraBundle/issues/282

**Comment by shubaivan** [comment]
yes, it's true, I try `"symfony/symfony": "~3.4",` and `"jms/di-extra-bundle": "~1.9"` ad still have `Call to a member function has() on null `

**Comment by Aerendir** [comment]
Same issue here: completely disabling the bundle solves the problem...

And yes, the problem is a known one not already solved as pointed out by @zhil that referenced the issue #282 .

Unfortunately are almost 7 months this issue exist and it is not fixed.

On my side I'm required to use this bundle to use JMSTranslationBundle, but as it completely breaks my application, I think I will switch to PHPTranslation that also offer a [Symfony's Bundle](https://github.com/php-translation/symfony-bundle).

Good luck.

**Comment by shubaivan** [comment]
thnks @Aerendir 
```
    AppBundle\Controller\:
        resource: '../../src/AppBundle/Controller'
        public: true
        tags: ['controller.service_arguments']
        calls:
            - [setContainer, ["@service_container"]]
```

**Comment by rvanlaak** [comment]
@shubaivan the problem with adding that call for all resources, is that not all of the controllers in '../../src/AppBundle/Controller' might have the `setContainer` method. 

In our case; 20% of the controllers are simple, easily autowirable callables and 80% are extending the base Controller class which means `setContainer` is available.

In other words, not sure if there is a generic solution for this. Maybe [`instantiateController`](https://github.com/schmittjoh/JMSDiExtraBundle/blob/master/HttpKernel/ControllerResolver.php#L76-L90) could make use of the FrameworkBundle's  [`ControllerResolver`](https://github.com/xabbuh/symfony/blob/master/src/Symfony/Bundle/FrameworkBundle/Controller/ControllerResolver.php)?
